packages/headless/src/features/facets/range-facets/date-facet-set/date-facet-actions.ts

```diff
@@ -127,27 +152,56 @@ export function validateManualDateRanges(
 
   options.currentValues.forEach((value) => {
     const {start, end} = buildDateRange(value);
-    if (dayjs(start).isAfter(dayjs(end))) {
+    if (dayjs(formatDate(start)).isAfter(dayjs(formatDate(end)))) {
       throw new Error(
-        `The start value is greater than the end value for the date range ${value.start} to ${value.end}`
+        `The start value is greater than the end value for the date range ${serializeDateRangeValue(
+          value.start
+        )} to ${serializeDateRangeValue(value.end)}`
       );
     }
   });
 }
 
+function formatRelativeDateValues(
```

Comment:
I used to use formatDate in both slices: DateFacetSet & RelativeDateSet, but this sometimes caused issues where the absolute date was slightly different (around 1 ms).
Instead I enhance the payload in order to have the map of absolute value + relative value that both slices can reuse. Single source of truth